packages/contracts-bedrock/src/dispute/FaultDisputeGame.sol

```diff
@@ -114,8 +119,15 @@ contract FaultDisputeGame is IFaultDisputeGame, Clone, ISemver {
         SPLIT_DEPTH = _splitDepth;
         GAME_DURATION = _gameDuration;
         VM = _vm;
+        WETH = _weth;
     }
 
+    /// @notice Receive function to allow the contract to receive ETH.
+    receive() external payable { }
+
+    /// @notice Fallback function to allow the contract to receive ETH.
+    fallback() external payable { }
```

Comment:
why do we need both?

Reply:
Solidity compiler complains if you don't have both, and we need fallback because proxy with immutable args throws extra calldata on the calls.